src/Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.cpp

```diff
@@ -184,15 +184,32 @@ void FixConservatives::operator()(
       const auto f_of_lorentz_factor = FunctionOfLorentzFactor{
           b_squared_over_d, tau_over_d, normalized_s_dot_b};
       const double upper_bound_of_lorentz_factor = 1.0 + tau_over_d;
-      const double lorentz_factor =
-          (equal_within_roundoff(lower_bound_of_lorentz_factor,
-                                 upper_bound_of_lorentz_factor)
-               ? lower_bound_of_lorentz_factor
-               :
-               // NOLINTNEXTLINE(clang-analyzer-core)
-               RootFinder::toms748(
-                   f_of_lorentz_factor, lower_bound_of_lorentz_factor,
-                   upper_bound_of_lorentz_factor, 1.e-14, 1.e-14, 50));
+
+      double lorentz_factor;
+
+      try {
+        lorentz_factor =
+            (equal_within_roundoff(lower_bound_of_lorentz_factor,
+                                   upper_bound_of_lorentz_factor)
+                 ? lower_bound_of_lorentz_factor
+                 :
+                 // NOLINTNEXTLINE(clang-analyzer-core)
+                 RootFinder::toms748(
+                     f_of_lorentz_factor, lower_bound_of_lorentz_factor,
+                     upper_bound_of_lorentz_factor, 1.e-14, 1.e-14, 50));
+      } catch (std::exception& exception) {
```

Comment:
[not yet a change request] For my understanding: `toms748` can throw different exceptions for different failure modes. Would it not be more useful to catch these separately and inform the user of the specific failure mode? Or, if we're confident in the inputs so that we should never see a `domain_error` here, then why not just catch the `convergence_error` ?

Reply:
What if we also stream out `exception.what()`? That should contain such info and doesn't require multiple catch blocks

Reply:
That sounds like a useful improvement with minimal added complexity 👍